medusa/providers/torrent/html/gimmepeers.py

```diff
@@ -0,0 +1,173 @@
+from medusa.providers.torrent.torrent_provider import TorrentProvider
+
+from requests.compat import urljoin
+from requests.utils import dict_from_cookiejar
+
+log = BraceAdapter(logging.getLogger(__name__))
+log.logger.addHandler(logging.NullHandler())
+
+
+class GimmePeersProvider(TorrentProvider):
+    """GimmePeers Torrent provider."""
+
+    def __init__(self):
+
+        TorrentProvider.__init__(self, 'GimmePeers')
+
+        self.url = 'https://www.gimmepeers.com'
+        self.urls = {
+            'login': urljoin(self.url, 'takelogin.php'),
+            'search': urljoin(self.url, 'browse.php'),
+            'download': urljoin(self.url, '%s'),
+        }
+
+        self.username = None
+        self.password = None
+
+        self.cache = tv.Cache(self)
+
+        self.search_params = {
+            'c20': 1,
+            'c21': 1,
+            'c25': 1,
+            'c24': 1,
+            'c23': 1,
+            'c22': 1,
+            'c1': 1
+        }
+
+    def search(self, search_strings, age=0, ep_obj=None, **kwargs):  # pylint: disable=too-many-locals
+        """
+        Search a provider and parse the results.
+
+        :param search_strings: A dict with mode (key) and the search value (value)
+        :param age: Not used
+        :param ep_obj: Not used
+        :returns: A list of search results (structure)
+        """
+        results = []
+        if not self.login():
+            return results
+
+        for mode in search_strings:
+
+            log.debug(u'Search Mode: {0}'.format(mode))
+            for search_string in search_strings[mode]:
+                if mode != 'RSS':
+                    log.debug(u'Search string: {0}'.format(search_string))
+
+                self.search_params['search'] = search_string
+
+                data = self.session.get(self.urls['search'], params=self.search_params)
+                if not data:
+                    continue
+
+                results += self.parse(data.text, mode)
+
+        return results
+
+    def parse(self, data, mode):
+        """
+        Parse search results for items.
+
+        :param data: The raw response from a search
+        :param mode: The current mode used to search, e.g. RSS
+        :return: A list of items found
+        """
+        items = []
+
+        with BS4Parser(data, 'html5lib') as html:
+            torrent_table = html.find('table', class_='browsetable')
+            torrent_rows = torrent_table('tr') if torrent_table else []
+
+            # Continue only if one Release is found
+            if len(torrent_rows) < 2:
+                log.debug(u'Data returned from provider does not contain any torrents')
+                return items
+
+            for result in torrent_rows[1:]:
+                try:
+                    cells = result('td')
+
+                    link = cells[1].find('a')
+                    download_url = self.urls['download'] % cells[2].find('a')['href']
```

Comment:
Ill correct my previous comment. You can put the urljoin right here. No need to template the string

Reply:
all fixed up, replaced it with urljoin